### ISSUE TITLE
basins_2D is unreasonably slow

### ISSUE THREAD

**Issue by Datseris** [JuliaDynamics/ChaosTools.jl#183]
MWE:
```julia
using DynamicalSystems

# Henon global bifurcation
he = Systems.henon(;a = 1.4, b = 0.3)

Z = 300
xg = range(-1.5, 1.5; length = Z)
yg = range(-0.5, 0.5; length = Z)
@time basins, attractors = basins_2D(xg, yg, integrator(he); mc_bas = 3)
```

```
  4.711121 seconds (58.35 M allocations: 1.551 GiB, 5.56% gc time)
```

This large amount of allocations, 58 million (!!!) is definitely an error. I see no reason why we should be allocating so much memory all the time. My guess is that we are creating a lot of vectors in the algorithm that shouldn't exist.

cc @awage

**Comment by awage** [comment]
I think I know what is going on. It should be easy to fix.

**Comment by awage** [comment]
Some improvements:
```
@time basins, attractors = basins_2D(xg, yg, integrator(he))
0.770388 seconds (9.51 M allocations: 403.362 MiB, 8.06% gc time)
```

**Comment by Datseris** [comment]
Can you add the fix to the open #184 ?

**Comment by awage** [comment]
We can close #183, fixed in #184
